diff -uNr anaconda-19.31.79__orig/pyanaconda/installclasses/clearos.py anaconda-19.31.79/pyanaconda/installclasses/clearos.py
--- anaconda-19.31.79__orig/pyanaconda/installclasses/clearos.py	2014-06-30 17:13:18.111386761 +0100
+++ anaconda-19.31.79/pyanaconda/installclasses/clearos.py	2014-06-30 17:13:43.071386726 +0100
@@ -47,7 +47,7 @@
 
     _l10n_domain = "comps"
 
-    efi_dir = "redhat"
+    efi_dir = "clearos"
 
     def setDefaultPartitioning(self, storage):
         autorequests = [PartSpec(mountpoint="/", fstype=storage.defaultFSType,
